### Задача 1. 
Из колоды в 52 карты извлекаются случайным образом 4 карты. 

a) Найти вероятность того, что все 4 карты – крести. 

Вероятность события выичсляется как отношение m/n. Где m - число благоприятствующих исходов, n - общее число исходов.
Так как порядок карт с мастью крести нам не важен, используем формулу сочетания.

from math import factorial
def combinations(k, n1):
    return factorial(n1)/(factorial(k) * factorial(n1 - k))

Вычислим m. Так как всего карт 52 а масти 4, получается что на одну масть приходится 13 карт. k = 4, так как балгоприятствующий исход - когда все 4 карты будут с мастью крести, а n1 = 13, так как выбирать мы должны из одной масти.

In [15]:
m = combinations(4, 13)

In [16]:
m

715.0

Вычислим n. Так как всего карт 52 а масти 4, получается что на одну масть приходится 13 карт. Следовательно n = k/n1, где k = 4, так как всего у нас 4 масти в колоде, а n1 - 52, так как всего 52 карты в колоде.

In [17]:
n = combinations(4, 52)

In [18]:
n

270725.0

Тогда вероятность того, что все 4 карты окажутся крести будет равна отношению:

In [19]:
m/n

0.0026410564225690276

P = 0.3 %

б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.

Сумма вероятности противоположных событий равна 1. А значит P1 + P2 = 1, где P1 - вероятность того, что среди 4-х карт окажется хотя бы один туз, P2 - вероятность того что среди 4-х карт не будет ни одного туза. 
А значит P1 = 1 - P2

Найдем P2 по формуле P = m/n

In [55]:
from math import factorial
def combinations(k, n1):
    return factorial(n1)/(factorial(k) * factorial(n1 - k))

Вычислим m.  где k = 4, а n1 = 48, так как благоприятствующий исход когда мы не вытянем ни одного туза из колоды, то есть когда наш выбор будепт сделан из 52 - 4 = 48 карт.

In [30]:
m = combinations(4, 48)

In [31]:
m

194580.0

Вычислим n. где k = 4, n1 = 52 так как в общем случае мы тянем 4 карты из полной колоды. 

In [32]:
n = combinations(4, 52)

In [33]:
n

270725.0

Тогда вероятность того, что мы не вытянем ни одного туза:

In [49]:
P2 = m/n

In [50]:
P2

0.7187367254594146

А вероятность что среди 4-х карт окажется хотя бы один туз будет равна:

In [51]:
P1 = 1 - m/n

In [52]:
P1

0.2812632745405854

P1 = 28.1 %

### Задача 2. 
На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно . Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?

P = m/n

m = 1, так как есть всего лишь 1 благоприятный исход - человек откроет дверь с первой попытки. 
А вот n будем считать по формуле сочентания, так как порядок цифр нам не важен потому что нажимать их нужно одновременно:

In [54]:
from math import factorial
def combinations(k, n1):
    return factorial(n)/(factorial(k) * factorial(n - k))

где k = 3, так как человеку нужно выбрать и нажать три цифры, а n1 = 10, так как цифр на замке десять. 

In [60]:
n = combinations(3, 10)

In [61]:
n

120.0

In [70]:
P = 1/n

In [71]:
P

0.008333333333333333

Вероятность открытия замка с первой попытки составляет P = 0.83 %

### Задача 3.
В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все 3 извлеченные детали окрашены?

P = m/n, где m и n считаем по формуле сочетания, так как порядок нам не важен:

In [75]:
from math import factorial
def combinations(k, n1):
    return factorial(n1)/(factorial(k) * factorial(n1 - k))

Рассчитаем m. Количество окрашенных деталей 9, а значит в благоприятном случае мы должны выбрать все 3 детали из этих 9. Следовательно k = 3, n1 = 9

In [76]:
m = combinations(3, 9)

In [77]:
m

84.0

Рассчитаем n. Количество всех деталей 15, а значит в общем случае мы будем выбирать 3 детали из всех 15. Следовательно k = 3, n1 = 15

In [78]:
n = combinations(3, 15)

In [79]:
n

455.0

In [80]:
P = m/n

In [81]:
P

0.18461538461538463

Вероятность достать 3 окрашенных детали равна 18.5 %

### Задача 4. 
В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?

P = m/n, где m и n считаем по формуле сочетания, так как порядок нам не важен:

In [82]:
from math import factorial
def combinations(k, n1):
    return factorial(n1)/(factorial(k) * factorial(n1 - k))

m = 1, так как у нас только один благоприятный исход при котором достанем два выигрышных билета из двух.

Рассчитаем n. k = 2, n1 = 100, так как в общем случае тянем два билета из всех возможных вариантов.

In [83]:
n = combinations(2, 100)

In [84]:
n

4950.0

In [85]:
P = 1/n

In [86]:
P

0.00020202020202020202

Вероятность вытянуть два выигрышных билета равна 0.02 %